inference-explainers/inference_api.md

```diff
@@ -0,0 +1,252 @@
+    {
+      "model_path": "pcvr_v1",
+      "checksum": "dd94b3b08ea19b4240aa5e5f68ff0447b40e37ebdd06cc76fa1a2cf61143a10c",
+      "warm_up_batch_request_json": "..."
+    },
+    {
+      "model_path": "pcvr_v2",
+      "checksum": "...",
+      "warm_up_batch_request_json": "..."
+    }
+  ]
+}
+```
+
+This model configuration file features a top-level array with each entry containing the metadata of the fetched models. The mandatory **model_path** field specifies the path to the model in the cloud storage. The optional **checksum** field is the SHA256 checksum of the model represented as a hexadecimal string.
+
+Model checksums are computed using the following steps:
+
+1. Compute the SHA256 checksum for each individual model file.
+2. Arrange the file checksums in ascending order based on their file paths.
+3. Concatenate all the ordered file checksums into a single string.
+4. Compute the final SHA256 checksum on the concatenated checksum string.
+
+The listed steps are equivalent to the following bash command:
+
+```
+find <model_path> -type f -exec sha256sum {} \; | sort -k 2 | awk '{print $1}' | tr -d '\n' | sha256sum | awk '{print $1}'
+```
+
+The optional **warm_up_batch_request_json** is a JSON batch request used to warm up the model before any traffic is served. The format of this warm-up request uses the same format as the input to the runInference JavaScript function, which is described later in this document.
+
+Refer to the [B&A Inference Onboarding Guide][4] for more details about using the model configuration file with the Terraform configuration.
+
+## JavaScript API
+
+We expose the Inference service capabilities as JavaScript functions in ad tech code modules. Currently, the Inference service can be executed as a part of the proprietary bidding logic. Ad techs can query available models and issue batched inference service requests to a selected model set.
+
+### JavaScript functions
+
+#### getModelPaths
+
+The `getModelPaths` function doesn’t accept arguments and returns the available models in the inference service sidecar. The returned result is represented in a serialized JSON array of model paths. For example, [“pcvr_v1”, “pcvr_v2”].
+
+#### runInference
+
+The `runInference` function accepts batch requests and returns batch responses.
+
+#### Batch request
+
+The batch inference service request is represented as a JSON object containing a top-level array. Each element in the array represents an individual request and is represented as a JSON object.
+
+#### Individual request mandatory fields
+
+- **model_path:** Specifies the target model for the request
+- **tensors:** An array of JSON objects representing the input tensors to the model.
+
+#### Tensor mandatory fields
+
+- **data_type:** Specifies the data type of the tensor, chosen from DOUBLE, FLOAT, INT8, INT16, INT32, INT64.
+- **tensor_shape:** An array specifying the shape of the tensor.
+- **tensor_content:** A flattened representation of the tensor values.
```

Comment:
wonder if we should say "flattened row major representation .."

Reply:
Done